tests/test_clang_delta.py

```diff
@@ -32,3 +35,289 @@ async def equals_source(x):
+        return True
+
+    # Simple code that has instances counted but fails during apply
+    source = b"int main() { return 0; }"
+    problem = BasicReductionProblem(
+        source, is_interesting, work=WorkContext(parallelism=1)
+    )
+
+    # Should not raise, should return the original
+    result = await pump(problem)
+    assert result == source
+
+
+# =============================================================================
+# Additional edge case tests for complete coverage
+# =============================================================================
+
+
+def test_find_clang_delta_when_found_in_path():
+    """Test find_clang_delta returns path when found via which()."""
+    from unittest.mock import patch
+
+    # Mock which to return a path
+    with patch(
+        "shrinkray.passes.clangdelta.which", return_value="/usr/bin/clang_delta"
+    ):
+        result = find_clang_delta()
+        assert result == "/usr/bin/clang_delta"
+
+
+def test_find_clang_delta_when_not_in_path():
+    """Test find_clang_delta falls back to glob when not in PATH."""
+    from unittest.mock import patch
+
+    # Mock which to return None (not in PATH)
+    with patch("shrinkray.passes.clangdelta.which", return_value=None):
+        # This will use glob to find clang_delta
+        result = find_clang_delta()
+        # Result depends on whether clang_delta is installed via homebrew/apt
+        # We just verify it doesn't crash and returns something (empty string if not found)
+        assert isinstance(result, str)
+
+
+def test_find_clang_delta_when_not_found_anywhere():
+    """Test find_clang_delta returns empty string when not found."""
+    from unittest.mock import patch
+
+    # Mock both which and glob to return nothing
+    with patch("shrinkray.passes.clangdelta.which", return_value=None):
+        with patch("shrinkray.passes.clangdelta.glob", return_value=[]):
+            result = find_clang_delta()
+            assert result == ""
+
+
+async def test_query_instances_raises_clang_delta_error():
+    """Test query_instances raises ClangDeltaError for non-assertion errors.
+
+    This tests line 133: raise ClangDeltaError(msg) when CalledProcessError
+    doesn't contain 'Assertion failed'.
+    """
+    import subprocess
+    from unittest.mock import patch
+
+    cd = ClangDelta(find_clang_delta())
+
+    # Mock trio.run_process to raise CalledProcessError without "Assertion failed"
+    error = subprocess.CalledProcessError(1, "clang_delta")
+    error.stdout = b"Some other error"
+    error.stderr = b"More error info"
+
+    with patch("trio.run_process", side_effect=error):
+        with pytest.raises(ClangDeltaError):
+            await cd.query_instances("rename-var", b"int main() {}")
+
+
+async def test_apply_transformation_no_modification():
+    """Test apply_transformation returns original when 'No modification' error.
+
+    This tests line 167: return data when the error message indicates no modification.
```

Comment:
Line 167 is not a durable referenc do you mean test_pump_handles_assertion_failure()